test/functional/sqlsrv/sqlsrv_ae_datetimes_as_strings.phpt

```diff
@@ -0,0 +1,595 @@
+                              $month_name." ".$day.", ".$year,
+                              $day."-".$month_name."-".$year,
+                              $day."-".$month_abbr."-".$year
+                              );
+$time_formats_nonODBC = array($hour12.":".$minute." ".$meridian,
+                              $hour12.":".$minute.":".$second." ".$meridian,
+                              $hour12.":".$minute.":".$second.".".$frac." ".$meridian,
+                              $hour12.":".$minute.":".$second.".".$frac.$frac2." ".$meridian,
+                              $hour.":".$minute
+                              );
+
+// Create arrays containing the ODBC-standard formats, and larger arrays
+// containing the non-standard formats, for the supported SQL Server
+// date and time types.
+$date_formats_all = array_merge($date_formats, $date_formats_nonODBC);
+$time_formats_all = array_merge($time_formats, $time_formats_nonODBC);
+
+$datetime_formats_all = array();
+$datetime2_formats_all = array();
+$datetimeoffset_formats_all = array();
+$datetimesmall_formats_all = array();
+
+$SZ_TIME_all = sizeof($time_formats_all);
+$SZ_DATE_all = sizeof($date_formats_all);
+$SZ_DATETIME_all = $SZ_TIME_all*$SZ_DATE_all;
+
+// Create compound date/time types. For datetime, remove the extra precision
+// of $frac2. For smalldatetime, remove the extra precision of $frac and $frac2.
+// If the data in $frac and/or $frac2 is found elsewhere in the date/time, the
+// data will be garbled. For example, if the year is 2002 and $frac2 is 2002, 
+// the code below will remove any instances of '2002' in the datetime and
+// smalldatetime strings, producing garbage for those types. User must be
+// cognizant of this when testing different dates and times.
+for ($i=0; $i<$SZ_DATE_all; $i++)
+{
+    for ($j=0; $j<$SZ_TIME_all; $j++)
+    {
+        $datetime_formats_all[] = str_replace($frac2, "", $date_formats_all[$i]." ".$time_formats_all[$j]);
+        $datetime2_formats_all[] = $date_formats_all[$i]." ".$time_formats_all[$j];
+        $datetimeoffset_formats_all[] = $date_formats_all[$i]." ".$time_formats_all[$j].$tz_correction;
+        if (str_replace(".".$frac.$frac2, "", $date_formats_all[$i]." ".$time_formats_all[$j]) == ($date_formats_all[$i]." ".$time_formats_all[$j])) {
+            $datetimesmall_formats_all[] = str_replace(".".$frac, "", $date_formats_all[$i]." ".$time_formats_all[$j]);
+        }
+        else {
+            $datetimesmall_formats_all[] = str_replace(".".$frac.$frac2, "", $date_formats_all[$i]." ".$time_formats_all[$j]);
+        }
+    }
+}
+
+date_default_timezone_set('Canada/Pacific');
+sqlsrv_configure('WarningsReturnAsErrors', 0);
+sqlsrv_configure('LogSeverity', SQLSRV_LOG_SEVERITY_ALL);
+sqlsrv_configure('LogSubsystems', SQLSRV_LOG_SYSTEM_OFF);
+
+require_once('MsCommon.inc');
+
+$returnDatesAsStrings = true;
+
+$conn = AE\connect(array('ReturnDatesAsStrings' => $returnDatesAsStrings));
```

Comment:
Suggestion: refactor this last section into one method that takes the boolean flag $returnDateAsStrings